systemd/ua-premium.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Ubuntu Advantage premium attach
+
+[Service]
+Type=oneshot
+ExecStartPre=/usr/bin/cloud-init status --wait
```

Comment:
I agree with @basak, we should be using native systemd dependencies to get this to start at the appropriate time. This approach will add another invocation of cloud-init to each boot, which is a performance hit we don't need to take.

Reply:
+1 shifting back to the earlier implementation which goes as soon as cloud-init has detected the datasource and brought up 'final' networking.

[Install]
WantedBy=multi-user.target
# cloud-init has completed datasource detection by network-online target
After=network-online.target
Want=network-online.target